### PR TITLE
feat: Implements the parseEvent function so that it can parse a valid push event and return the corresponding CI Job.

### LINKED ISSUE #24: Implement parseEvent in WebHookHandler
so that it can parse HTTP requests and return the ContinuousIntegrationJob data structure

### PR DESCRIPTION
**Pull request by 641bill**
Fixes #24, please also help to link this issue with the new commit from squash and merge.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,10 +15,14 @@ repositories {
 
 dependencies {
     implementation 'javax.servlet:javax.servlet-api:4.0.1'
-    implementation 'org.eclipse.jetty:jetty-server:11.0.8'
     testImplementation 'junit:junit:4.+'
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.2'
     testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.8.2'
+    implementation 'org.json:json:20211205'
+    implementation 'org.springframework:spring-test:5.3.15'
+    implementation 'org.eclipse.jetty:jetty-server:9.4.43.v20210629'
```
Comment (edv1n) [inline build.gradle:23]: any specific reason why we pick an older version of jetty?

Comment (641bill) [reply build.gradle:23]: Originally after I removed the requirement of java 11 the previous newer version of jetty won't compile using gradle and throws error: [Unable to compile: class file has wrong version 55.0, should be 52.0], and after I lowered the version of it, it builds

File: build.gradle [hunk 2]
```diff
@@ -15,10 +15,14 @@ repositories {
...
+}
+
+test {
+    useJUnitPlatform()
 }
-sourceCompatibility = JavaVersion.VERSION_11
```
Comment (edv1n) [inline build.gradle:23]: let's not remove this in this PR.

File: build.gradle [hunk 3]
```diff
@@ -15,10 +15,14 @@ repositories {
...
-targetCompatibility = JavaVersion.VERSION_11
```
Comment (edv1n) [inline build.gradle:24]: let's not remove this in this PR.

**Pull request merged**